### DISCUSSION TITLE
Mechanics/Behaviour of Next/Link + SSR + getServerSideProps

### DISCUSSION THREAD

**Discussion by RapidOwl** [vercel/next.js#28477] [Help] [answered]
I'm a little confused as to how SSR with Next/Link is expected to work. I'm running the Next tutorial app and have added another page that gets some data from the Dog API.

When I load the initial page (without `getServerSideProps`) it is rendered before it hits the browser. When I navigate to the other page using the Next/Link component, it downloads a bunch of JavaScript and a JSON document that contains the data I requested in `getServerSideProps`.

I was under the impression that use of `getServerSideProps` would force a page to render on the server. Is this not the case?

I was expecting this to do something along the lines of TurboLinks where the main content is replaced by content from the server and the `head` is merged.

**Comment by jamesmosier** [comment, marked as answer]
When you use `getServerSideProps` and do a full page refresh (i.e. reload the browser or navigate directly to that page), the page is fully rendered on the server. If you navigate to that same page via `next/link`, instead of fully re-rendering the entire page, Next makes a `fetch` call to an internal API that executes your `getServerSideProps` code and returns that `props` as `json`. This is so that the transition to the next page is faster because your browser doesn't have to redownload all of the static assets (js, css, doc, etc.) again for the site and instead just has to make a fetch call to run your logic and then React renders those props and the page content.

The [docs](https://nextjs.org/docs/basic-features/data-fetching#only-runs-on-server-side) also explain this concept a bit.

**Reply by RapidOwl** [reply]
Aha! Thanks for clarifying.
